pkg/controller/gkenetworkparamset/gnpcontroller_validations.go

```diff
@@ -233,3 +239,77 @@ func crossValidateNetworkAndGnp(network *networkv1.Network, params *networkv1.GK
 		IsValid: true,
 	}
 }
+
+// hasNewPodRange returns true if the node has new Pod range that's not in the "default" params
+func (c *Controller) hasNewPodRange(node *v1.Node) bool {
+	defaultPodRanges, err := c.getParamsPodRanges(networkv1.DefaultPodNetworkName)
+	if err != nil {
+		klog.V(4).Infof("check new Pod range on node %q error: %v", node.Name, err)
+		return false
+	}
+	v, ok := node.Labels[utilnode.NodePoolPodRangeLabelPrefix]
+	// here assumes the pod ranges can not overlapped
```

Comment:
do we validate this somewhere?

Reply:
Do you mean the `pod ranges can not overlapped` part? It would be validate in the cluster server when the range is added to the cluster. If the range is overlapped with the existing ranges, it will fails right away, otherwise the node will pick it and put it into the label. 

(Leave the comment as unresolved if other reviewers have the same concern)